src/shogun/evaluation/MachineEvaluation.cpp

```diff
@@ -121,5 +161,22 @@ CMachine* CMachineEvaluation::get_machine() const
 
 EEvaluationDirection CMachineEvaluation::get_evaluation_direction()
 {
-	return m_evaluation_criterion->get_evaluation_direction();
+	REQUIRE(m_evaluation_criteria->get_array_size()==1,
+		"Multiple Metrics provided. Please use get_evaluation_directions()");
+	CEvaluation* temp_eval = (CEvaluation*) m_evaluation_criteria->get_element_safe(0);
+	return temp_eval->get_evaluation_direction();
+}
+
+CDynamicArray<EEvaluationDirection>* CMachineEvaluation::get_evaluation_directions()
+{
+	CDynamicArray<EEvaluationDirection>* list_evaluation_directions = new CDynamicArray<EEvaluationDirection>();
+	int32_t num_eval = m_evaluation_criteria->get_num_elements();
+	for (int32_t i = 0; i < num_eval;i++)
+	{
+		//following line is type safe which is why we don't check
+		CEvaluation* temp_eval = (CEvaluation*) m_evaluation_criteria->get_element_safe(i);
+		list_evaluation_directions->push_back(temp_eval->get_evaluation_direction());
+	}
+	SG_REF(list_evaluation_directions);
+	return list_evaluation_directions;
```

Comment:
SWIG needs to know that this method returns a new object, see Evaluation.i interfaces/modular